In [1]:
# importing Libraries
import pandas as pd
import numpy as np
import os

### 1. Import the data sets into Jupyter

In [22]:
#  Tell Python to remember a main folder path
path = r'/Users/moneymakingmitch/Downloads/Instacart Basket Analysis/02 Data/Original Data'

In [55]:
#  Import dataset orders_products_prior.csv
df_ords_prior = pd.read_csv(os.path.join(path,'orders_products_prior.csv'), index_col = False)

In [59]:
# Import dataset orders.csv

df_ords = pd.read_csv(os.path.join(path,'orders.csv'), index_col = False)

### 2. Check the dimension of the imported dataframes

In [60]:
# check the output
df_ords_prior.head()

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [61]:
# check the output
df_ords.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [62]:
df_ords_prior.shape

(32434489, 4)

In [63]:
df_ords.shape

(3421083, 7)

### 3. Merge the dataframes

In [64]:
df_merged_large = df_ords.merge(df_ords_prior, on = 'order_id', indicator = True)

In [65]:
df_merged_large.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2539329        1    prior             1          2                  8   
2   2539329        1    prior             1          2                  8   
3   2539329        1    prior             1          2                  8   
4   2539329        1    prior             1          2                  8   

   days_since_prior_order  product_id  add_to_cart_order  reordered _merge  
0                     NaN         196                  1          0   both  
1                     NaN       14084                  2          0   both  
2                     NaN       12427                  3          0   both  
3                     NaN       26088                  4          0   both  
4                     NaN       26405                  5          0   both

### 4. Use the indicator argument to check whether there was a full match between the two dataframes

In [66]:
df_merged_large['_merge'].value_counts()

_merge
both          32434489
left_only            0
right_only           0
Name: count, dtype: int64

#### The resulting dataframe (after the merge) has 32,434,489 rows, and each of those rows have information found in both input data sets, as we used an inner join for the purposes of this project

### 5. Export data in pickle format

In [67]:
# Export data to csv

df_merged_large.to_csv(os.path.join(path,r'/Users/moneymakingmitch/Downloads/Instacart Basket Analysis/02 Data/Prepared Data/','orders_products_combined.csv'))

In [68]:
# Export data to pkl

df_merged_large.to_pickle(os.path.join(path,r'/Users/moneymakingmitch/Downloads/Instacart Basket Analysis/02 Data/Prepared Data/','orders_products_combined.pkl'))